evalml/tests/automl_tests/test_iterative_algorithm.py

```diff
@@ -95,18 +96,37 @@ def test_iterative_algorithm_results(dummy_binary_pipeline_classes):
 
     # subsequent batches contain pipelines_per_batch copies of one pipeline, moving from best to worst from the first batch
     last_batch_number = algo.batch_number
+    last_pipeline_number = algo.pipeline_number
     all_parameters = []
-    while algo.batch_number < 10:
-        next_batch = algo.next_batch()
-        assert len(next_batch) == algo.pipelines_per_batch
-        cls = dummy_binary_pipeline_classes[(algo.batch_number - 2) % len(dummy_binary_pipeline_classes)]
-        assert [p.__class__ for p in next_batch] == [cls] * len(next_batch)
-        assert algo.pipeline_number == len(dummy_binary_pipeline_classes) + (algo.batch_number - 1) * algo.pipelines_per_batch
-        assert algo.batch_number == last_batch_number + 1
-        last_batch_number = algo.batch_number
-        print([p.parameters for p in next_batch])
-        all_parameters.extend([p.parameters for p in next_batch])
-        scores = -np.arange(0, len(next_batch))
-        for score, pipeline in zip(scores, next_batch):
-            algo.add_result(score, pipeline)
-    assert any([p != dummy_binary_pipeline_classes[0]({}).parameters for p in all_parameters])
+
+    for i in range(1, 5):
+        for _ in range(len(dummy_binary_pipeline_classes)):
+            next_batch = algo.next_batch()
+            assert len(next_batch) == algo.pipelines_per_batch
+
+            cls = dummy_binary_pipeline_classes[(algo.batch_number - 2) % len(dummy_binary_pipeline_classes)]
+            assert [p.__class__ for p in next_batch] == [cls] * len(next_batch)
+            assert algo.pipeline_number == last_pipeline_number + len(next_batch)
+            last_pipeline_number = algo.pipeline_number
+            assert algo.batch_number == last_batch_number + 1
+            last_batch_number = algo.batch_number
+            print([p.parameters for p in next_batch])
```

Comment:
extra print

Reply:
I think @dsherry originally added the print statement cause it added clarity, so just keeping that in here :D